ThirdPartyAdapters/unity/unity/src/main/java/com/google/ads/mediation/unity/UnityAdapter.java

```diff
@@ -193,20 +127,91 @@ public void requestInterstitialAd(Context context,
         Activity activity = (Activity) context;
         mActivityWeakReference = new WeakReference<>(activity);
 
-        UnitySingleton.getInstance().initializeUnityAds(activity, gameId);
-        UnitySingleton.getInstance().loadAd(mUnityAdapterDelegate);
+        UnityInitializer.getInstance().initializeUnityAds(activity, gameId, new IUnityAdsInitializationListener() {
+            @Override
+            public void onInitializationComplete() {
+                Log.d(UnityAdapter.TAG, "Unity Ads successfully initialized");
+                loadInterstitialAd(mPlacementId);
```

Comment:
why do you do it this way?

can you call `loadInterstitialAd` right after ` UnityInitializer.getInstance().initializeUnityAds`?

Reply:
Yes, this is the way facebook did it and i thought the idea was that it should only call load if initialize is successful

Reply:
this should be fine. we can support both cases.